### PR TITLE
Update sui-breaking-changes.md

### PR DESCRIPTION
**Pull request by randall-Mysten**
## Description 

Adds additional Breaking Changes entries to the topic

## Test Plan 

Manually

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: doc/src/doc-updates/sui-breaking-changes.md [hunk 1]
```diff
@@ -32,4 +36,8 @@ This release changes the field value type in `TransactionData` from `gas_payment
...
+
+**[Major breaking change]** - We rewrote ****the ID leak verifier that governs usage of `UID`s in Sui Move code to to flip the logic. With this change, new objects require “fresh” `UID`s created in the function that creates the object. However, when the struct of an object is destroyed, the `UID` can be stored as if the object was wrapped (but without its contents). In contrast, the previous rules stated that the `UID` could come from anywhere, but required destruction when the object was unpacked. We made this change to make using dynamic fields a bit more efficient - you do not always need a `Bag` or `Table` if to retain access to dynamic fields after unpacking an object into its constituent fields. See https://github.com/MystenLabs/sui/pull/8026 for details and a migration example.
```
Comment (ronny-mysten) [inline doc/src/doc-updates/sui-breaking-changes.md:43]: "...We rewrote ****the ID..." Intentional?

File: doc/src/doc-updates/sui-breaking-changes.md [hunk 2]
```diff
@@ -4,15 +4,19 @@ title: Sui Breaking Changes in Release .28
...
+**[Major breaking change]** The `sui_getObject` endpoint now takes an additional configuration parameter of type `SuiObjectDataOptions` to control which fields the endpoint retrieves. By default, the endpoint retrieves only object references unless the client request  explicitly specifies other data, such as `type`, `owner`, or `bcs`. To learn more, see (https://github.com/MystenLabs/sui/pull/8817)
+
+**[Major breaking change]** - The ID leak verifier that governs usage of `UID`s in Sui Move code has been rewritten and flipped. New objects must now get “fresh” `UID`s created in the function where the object is made, but when the object’s struct destroyed, the UID can be stored as if the object was wrapped (but without it's contents). In contrast, the previous rules stated that the `UID` could come from anywhere, but must have been destroyed when the object was unpacked. We have made this change to make using dynamic fields a bit more ergonomic, so you do not always need a Bag or Table if you want to retain access to dynamic fields after unpacking an object into its constituent fields. See [PR 8026](https://github.com/MystenLabs/sui/pull/8026) for details and a migration example.
```
Comment (ronny-mysten) [inline doc/src/doc-updates/sui-breaking-changes.md:11]:
```suggestion
**[Major breaking change]** - The ID leak verifier that governs usage of `UID`s in Sui Move code has been rewritten and flipped. New objects must now get “fresh” `UID`s created in the function where the object is made, but when the object’s struct is destroyed, the UID can be stored as if the object was wrapped (but without it's contents). In contrast, the previous rules stated that the `UID` could come from anywhere, but must have been destroyed when the object was unpacked. We have made this change to make using dynamic fields a bit more ergonomic, so you do not always need a Bag or Table if you want to retain access to dynamic fields after unpacking an object into its constituent fields. See [PR 8026](https://github.com/MystenLabs/sui/pull/8026) for details and a migration example. 
```

File: doc/src/doc-updates/sui-breaking-changes.md [hunk 3]
```diff
@@ -4,15 +4,19 @@ title: Sui Breaking Changes in Release .28
 
 The next release of Sui, release 0.28, includes the breaking changes described in this topic. A breaking change is one that introduces new, or changed, Sui functionality that causes existing apps and implementations to stop functioning as expected. Updates to this topic for next week include information about how to resolve or update your project that might break as a result of these changes.
 
+New entries added 03/06.
+
+**[Major breaking change]** The `sui_getObject` endpoint now takes an additional configuration parameter of type `SuiObjectDataOptions` to control which fields the endpoint retrieves. By default, the endpoint retrieves only object references unless the client request  explicitly specifies other data, such as `type`, `owner`, or `bcs`. To learn more, see (https://github.com/MystenLabs/sui/pull/8817)
```
Comment (ronny-mysten) [inline doc/src/doc-updates/sui-breaking-changes.md:9]:
```suggestion
**[Major breaking change]** The `sui_getObject` endpoint now takes an additional configuration parameter of type `SuiObjectDataOptions` to control which fields the endpoint retrieves. By default, the endpoint retrieves only object references unless the client request  explicitly specifies other data, such as `type`, `owner`, or `bcs`. To learn more, see [PR 8817](https://github.com/MystenLabs/sui/pull/8817)
```

File: doc/src/doc-updates/sui-breaking-changes.md [hunk 4]
```diff
@@ -32,4 +36,8 @@ This release changes the field value type in `TransactionData` from `gas_payment
 
 **[API breaking change]** - To reduce the size of Sui Full node synchronization payloads, this release removes events from `TransactionEffect`. The events are still included in the `SuiTransactionResponse` returned by `sui_getTransaction` and `sui_submitTransaction` endpoints. For more information, see [PR 7822](https://github.com/MystenLabs/sui/pull/7822).
 
-**[API breaking change]** - The `StakedSui` object now includes the ID of the staking pool, `pool_id`. For more information, see [PR 8371](https://github.com/MystenLabs/sui/pull/8371).
\ No newline at end of file
+**[API breaking change]** - The `StakedSui` object now includes the ID of the staking pool, `pool_id`. For more information, see [PR 8371](https://github.com/MystenLabs/sui/pull/8371).
+
+**[Major breaking change]** The `sui_getObject` endpoint now takes an additional configuration parameter of type `SuiObjectDataOptions` to control which fields the endpoint retrieves. By default, the endpoint retrieves only object references unless the client request  explicitly specifies other data, such as `type`, `owner`, or `bcs`. To learn more, see (https://github.com/MystenLabs/sui/pull/8817)
```
Comment (ronny-mysten) [inline doc/src/doc-updates/sui-breaking-changes.md:41]:
```suggestion
**[Major breaking change]** The `sui_getObject` endpoint now takes an additional configuration parameter of type `SuiObjectDataOptions` to control which fields the endpoint retrieves. By default, the endpoint retrieves only object references unless the client request  explicitly specifies other data, such as `type`, `owner`, or `bcs`. To learn more, see [PR 8817](https://github.com/MystenLabs/sui/pull/8817)
```

File: doc/src/doc-updates/sui-breaking-changes.md (hunk 1 above)
Comment (ronny-mysten) [inline doc/src/doc-updates/sui-breaking-changes.md:43]:
```suggestion
**[Major breaking change]** - We rewrote ****the ID leak verifier that governs usage of `UID`s in Sui Move code to to flip the logic. With this change, new objects require “fresh” `UID`s created in the function that creates the object. However, when the struct of an object is destroyed, the `UID` can be stored as if the object was wrapped (but without its contents). In contrast, the previous rules stated that the `UID` could come from anywhere, but required destruction when the object was unpacked. We made this change to make using dynamic fields a bit more efficient - you do not always need a `Bag` or `Table` if to retain access to dynamic fields after unpacking an object into its constituent fields. See [PR 8026](https://github.com/MystenLabs/sui/pull/8026) for details and a migration example.
```
For consistency.

Comment (randall-Mysten) [reply doc/src/doc-updates/sui-breaking-changes.md]: nope, Notion issue

**Review by ronny-mysten — APPROVED** [review]

**Pull request merged**